lib/peep/buckets/custom.ex

```diff
@@ -34,22 +34,95 @@ defmodule Peep.Buckets.Custom do
     end
   end
 
+  ## Bucket binary search
+
   defp bucket_for_ast(buckets) do
-    bucket_defns =
-      for {boundary, bucket_idx} <- Enum.with_index(buckets) do
-        quote do
-          def bucket_for(n, _) when n < unquote(boundary), do: unquote(bucket_idx)
-        end
+    int_buckets = int_buckets(buckets, nil, 0)
+
+    float_buckets =
+      buckets
+      |> Enum.map(&(&1 * 1.0))
+      |> Enum.with_index()
```

Comment:
You can use `with_index/2` and do it once, but please double check the Elixir version support.

Reply:
It is executed in compile time so I wouldn't worry too much about it